x-pack/plugins/infra/server/lib/adapters/metadata/elasticsearch_metadata_adapter.ts

```diff
@@ -133,6 +139,111 @@ export class ElasticsearchMetadataAdapter implements InfraMetadataAdapter {
       buckets,
     };
   }
+
+  public async getMetricsMetadataForServices(
+    req: InfraFrameworkRequest,
+    sourceConfiguration: InfraSourceConfiguration,
+    start: number,
+    end: number,
+    filterQuery?: ServiceMetadataQuery
+  ): Promise<InfraServiceMetadataMetricsBucket[]> {
+    const query = {
+      index: sourceConfiguration.metricAlias,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              ...createQueryFilterClauses(filterQuery),
+              {
+                range: {
+                  [sourceConfiguration.fields.timestamp]: {
+                    gte: start,
+                    lte: end,
+                  },
+                },
+              },
+            ],
+          },
+        },
+        size: 0,
+        aggs: {
+          metrics: {
+            terms: {
```

Comment:
Here is how I'm using composites in some prototype code, it's not pretty but it shows you the mechanics of how to fetch all the results using the `after_key`. Making this generic is tricky because of how you have to include the `after` key deep into the aggregation, it's not impossible but I just didn't put that much time into making it generic. Hopefully this is helpful...

```javascript
import { isEmpty } from 'lodash';
const shouldFetchMore = response => {
  const { groupBy } = response.aggregations;
  if (!groupBy) return false;
  const afterKey = groupBy.after_key || {};
  const lastBucket = last(groupBy.buckets);
  if (isEmpty(lastBucket) || isEmpty(afterKey)) return false;
  return isEqual(afterKey, lastBucket.key);
};

export function getEverything(client, body) {
  const responses = [];
  const request = key => {
    const requestBody = key
      ? {
        ...body,
        aggs: {
          ...body.aggs,
          groupBy: {
            ...body.aggs.groupBy,
            composite: {
              ...body.aggs.groupBy.composite,
              after: key,
            },
          },
        },
      }
      : body;
    return client('search', {
      body: requestBody,
      index: options.index,
    }).then(response => {
      responses.push(response);
      if (shouldFetchMore(response)) {
        return request(response.aggregations.groupBy.after_key);
      }
      return responses;
    });
  };
  return request();
}
```